model/riscv_sys_control.sail

```diff
@@ -310,7 +310,7 @@ function exception_handler(cur_priv : Privilege, ctl : ctl_result,
       then print_platform("ret-ing from " ^ to_str(prev_priv)
                           ^ " to " ^ to_str(cur_privilege));
 
-      prepare_xret_target(Supervisor) & pc_alignment_mask()
+      prepare_xret_target(Supervisor) & sign_extend(if misa[C] then 0b110 else 0b100)
```

Comment:
Please don't repeat this in two places. You use @Timmmm 's `align_pc` function from #593 .

Reply:
Is there a reason one should not just reuse legalize_xepc? Perhaps because we want to capture the fact that the LSB isn't ever set in this case, it's only the second bit that needs handling specially?